pkg/env/env.go

```diff
@@ -286,13 +287,21 @@ func (e *testEnv) getFinishActions() []action {
 
 func (e *testEnv) execFeature(ctx context.Context, t *testing.T, f types.Feature) context.Context {
 	featName := f.Name()
+	labels := f.Labels()
 
 	// feature-level subtest
 	t.Run(featName, func(t *testing.T) {
+		// skip if feature name does not match
 		if e.cfg.FeatureRegex() != nil && !e.cfg.FeatureRegex().MatchString(featName) {
 			t.Skipf(`Skipping feature "%s": name not matched`, featName)
 		}
 
+		// skip if labels does not match
+		// run tests if --labels values matches the feature labels
+		if e.cfg.Labels() != nil && !reflect.DeepEqual(labels, e.cfg.Labels()) {
```

Comment:
Deep equality will not work here because it requires the maps (compared) to be of the same length. That will not always be the case using the framework. For instance I may have a set of labels with 3 keys defined in a feature test. However, I only use a map with one entry with --labels argument.

The best thing to do is to loop through the defined labels and look for the provided label.

Reply:
I have updated this check by looping through the defined labels.